src/test/java/com/textadventure/MainTest.java

```diff
@@ -8,4 +14,25 @@
     public void testMain() {
         Assert.assertEquals(true, true);
     }
+    @Test
+    public void testOutputToken() {
+    	String inputTest = " W S North run walk D ";
+    	Lexer lexer = new Lexer();
+    	ArrayList<Token> tokens = lexer.lex(inputTest);
+    	String output = "";
+    	for (Token token : tokens)
+    		output += token.type + " ";
```

Comment:
Consider using StringBuilder instead of +=

`StringBuilder output = new StringBuilder();
    	for (Token token : tokens)
    		output.append(token.type).append(" ");`